In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

from google.colab import files
uploaded = files.upload()

Saving bbq_summer.csv to bbq_summer.csv


In [2]:
#Reading of csv file
bbq = pd.read_csv('bbq_summer.csv')
print(bbq)

     bundleID                starter                         maindishI  \
0           1  Fried Chicken Tenders                       BBQ Brisket   
1           2  Fried Chicken Tenders                       BBQ Brisket   
2           3  Fried Chicken Tenders                       BBQ Brisket   
3           4  Fried Chicken Tenders                       BBQ Brisket   
4           5  Fried Chicken Tenders                       BBQ Brisket   
..        ...                    ...                               ...   
379       380  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
380       381  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
381       382  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
382       383  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
383       384  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   

        maindishII                side              dessert  avg_rating  
0          Sausage      Mac and Chees

In [3]:
bbq

bundleID                starter                         maindishI  \
0           1  Fried Chicken Tenders                       BBQ Brisket   
1           2  Fried Chicken Tenders                       BBQ Brisket   
2           3  Fried Chicken Tenders                       BBQ Brisket   
3           4  Fried Chicken Tenders                       BBQ Brisket   
4           5  Fried Chicken Tenders                       BBQ Brisket   
..        ...                    ...                               ...   
379       380  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
380       381  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
381       382  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
382       383  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
383       384  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   

        maindishII                side              dessert  avg_rating  
0          Sausage      Mac and Cheese      Blondie Brownie       6.040  
1          Sausage      Mac and Cheese  Apple Pie a la Mode       9.290  
2          Sausage        Potato Salad      Blondie Brownie       6.200  
3          Sausage        Potato Salad  Apple Pie a la Mode       8.840  
4          Sausage  French Fry Platter      Blondie Brownie       8.240  
..             ...                 ...                  ...         ...  
379  Steak Sampler      Mac and Cheese  Apple Pie a la Mode       4.738  
380  Steak Sampler        Potato Salad      Blondie Brownie       4.768  
381  Steak Sampler        Potato Salad  Apple Pie a la Mode       6.248  
382  Steak Sampler  French Fry Platter      Blondie Brownie       7.138  
383  Steak Sampler  French Fry Platter  Apple Pie a la Mode       7.428  

[384 rows x 7 columns]

In [4]:
#Dropping of BundleID
bbq1=bbq.drop(['bundleID'],axis=1)

In [5]:
#Dataset post drop operation
bbq1

starter                         maindishI     maindishII  \
0    Fried Chicken Tenders                       BBQ Brisket        Sausage   
1    Fried Chicken Tenders                       BBQ Brisket        Sausage   
2    Fried Chicken Tenders                       BBQ Brisket        Sausage   
3    Fried Chicken Tenders                       BBQ Brisket        Sausage   
4    Fried Chicken Tenders                       BBQ Brisket        Sausage   
..                     ...                               ...            ...   
379  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker  Steak Sampler   
380  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker  Steak Sampler   
381  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker  Steak Sampler   
382  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker  Steak Sampler   
383  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker  Steak Sampler   

                   side              dessert  avg_rating  
0        Mac and Cheese      Blondie Brownie       6.040  
1        Mac and Cheese  Apple Pie a la Mode       9.290  
2          Potato Salad      Blondie Brownie       6.200  
3          Potato Salad  Apple Pie a la Mode       8.840  
4    French Fry Platter      Blondie Brownie       8.240  
..                  ...                  ...         ...  
379      Mac and Cheese  Apple Pie a la Mode       4.738  
380        Potato Salad      Blondie Brownie       4.768  
381        Potato Salad  Apple Pie a la Mode       6.248  
382  French Fry Platter      Blondie Brownie       7.138  
383  French Fry Platter  Apple Pie a la Mode       7.428  

[384 rows x 6 columns]

In [6]:
#checking for null values
bbq1.isnull().values.any()

False

In [7]:
#Sorting of the dataset in a descending manner based on avg_rating
bbq1.sort_values(by='avg_rating', ascending=False)

starter                         maindishI      maindishII  \
223  Sticky Chicken Tenders                       Pulled Pork    Lobster Roll   
81    Fried Chicken Tenders  Buffalo Chicken Macaroni Stacker    Lobster Roll   
33    Fried Chicken Tenders                       Pulled Pork    Lobster Roll   
40    Fried Chicken Tenders                       Pulled Pork  Beef Short Rib   
5     Fried Chicken Tenders                       BBQ Brisket         Sausage   
..                      ...                               ...             ...   
350   Jumbo Shrimp Cocktail                       BBQ Chicken  Beef Short Rib   
314   Jumbo Shrimp Cocktail                       Pulled Pork         Sausage   
76    Fried Chicken Tenders  Buffalo Chicken Macaroni Stacker         Sausage   
308   Jumbo Shrimp Cocktail                       BBQ Brisket   Steak Sampler   
64    Fried Chicken Tenders                       BBQ Chicken  Beef Short Rib   

                   side              dessert  avg_rating  
223      Mac and Cheese  Apple Pie a la Mode       9.940  
81         Potato Salad  Apple Pie a la Mode       9.940  
33         Potato Salad  Apple Pie a la Mode       9.940  
40   French Fry Platter      Blondie Brownie       9.930  
5    French Fry Platter  Apple Pie a la Mode       9.900  
..                  ...                  ...         ...  
350        Potato Salad      Blondie Brownie       3.180  
314        Potato Salad      Blondie Brownie       2.870  
76   French Fry Platter      Blondie Brownie       2.630  
308        Potato Salad      Blondie Brownie       2.008  
64   French Fry Platter      Blondie Brownie       1.690  

[384 rows x 6 columns]

In [9]:
#Obtaining dummies
bbq2 = pd.get_dummies(bbq1, drop_first=True, columns=['starter', 'maindishI', 'maindishII','side','dessert'])

In [10]:
bbq2.head()

avg_rating  starter_Fried Chicken Tenders  starter_Jumbo Shrimp Cocktail  \
0        6.04                              1                              0   
1        9.29                              1                              0   
2        6.20                              1                              0   
3        8.84                              1                              0   
4        8.24                              1                              0   

   starter_Sticky Chicken Tenders  maindishI_BBQ Chicken  \
0                               0                      0   
1                               0                      0   
2                               0                      0   
3                               0                      0   
4                               0                      0   

   maindishI_Buffalo Chicken Macaroni Stacker  maindishI_Pulled Pork  \
0                                           0                      0   
1                                           0                      0   
2                                           0                      0   
3                                           0                      0   
4                                           0                      0   

   maindishII_Lobster Roll  maindishII_Sausage  maindishII_Steak Sampler  \
0                        0                   1                         0   
1                        0                   1                         0   
2                        0                   1                         0   
3                        0                   1                         0   
4                        0                   1                         0   

   side_Mac and Cheese  side_Potato Salad  dessert_Blondie Brownie  
0                    1                  0                        1  
1                    1                  0                        0  
2                    0                  1                        1  
3                    0                  1                        0  
4                    0                  0                        1

In [11]:
#viewing of columns post generation of dummies
bbq2.columns

Index(['avg_rating', 'starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie'],
      dtype='object')

In [12]:
#assigning input and outut variables
X = bbq2[['starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie']]
y = bbq2['avg_rating']

In [13]:
from sklearn.linear_model import LinearRegression
from sklearn import metrics
regressor = LinearRegression()
regressor.fit(X, y)

LinearRegression()

In [14]:
#Intercept value
regressor.intercept_

7.140387500000002

In [15]:
#Coefficients for the dataset
coef_df = pd.DataFrame(regressor.coef_, X.columns, columns=['Coefficient'])
coef_df

Coefficient
starter_Fried Chicken Tenders                  0.133571
starter_Jumbo Shrimp Cocktail                 -0.630588
starter_Sticky Chicken Tenders                 0.339275
maindishI_BBQ Chicken                         -0.209208
maindishI_Buffalo Chicken Macaroni Stacker     0.536821
maindishI_Pulled Pork                          0.529312
maindishII_Lobster Roll                        1.048096
maindishII_Sausage                             0.331992
maindishII_Steak Sampler                      -0.500462
side_Mac and Cheese                            0.575125
side_Potato Salad                             -0.022194
dessert_Blondie Brownie                       -0.856204

In [16]:
regressor.coef_

array([ 0.13357083, -0.6305875 ,  0.339275  , -0.20920833,  0.53682083,
        0.5293125 ,  1.04809583,  0.33199167, -0.5004625 ,  0.575125  ,
       -0.02219375, -0.85620417])

Conjoinint analysis has been performed here for the choosing of food items provided by the vendor for summer grill menu. Recommendations for barbeque options are based on the consumer rating. The coefficients generated here provides better understanding of food preference and the strength of opinions of customers.On analysing the coefficients the model suggests:        
              

**Starter**- Sticky Chicken Tenders , Coefficient value - 0.339275 , Price- $2.9 

For Main Dish I, the model suggests Buffalo Chicken Macaroni Stacker of coefficient value 0.536821 and Price of $6.1 to be the best. But Pulled Pork is rated equally good by the customer rating. So we will be going with Pulled pork option for main dish I.

**Main Dish I**- Pulled Pork, Coefficient value - 0.529312, Price - $5.7

For Main Dish II, the model suggests Lobster Roll seems to be one of the best seller at Lobster Land.

**Main Dish II**- Lobster Roll, Coefficient value - 1.048096, Price - $5.2

**Side dish**- Mac and Cheese, Coefficient Value -0.575125 ,Price- $0.45

**Dessert**- Apple Pie a la Mode, with a positive Coefficient value, Price- $1.1

Above mentioned are the recommendations based on the analysis to Lobster land management.    
Total price - $15.35



As the Lobster Land management has decided to charge a flat $15 fee for all visitors who enter the Barbeque Tent at the Portland Calling festival the food recommendations are evident to attract more customers thereby maximising the revenue and foot fall at the festival.